File: Python/246. Strobogrammatic Number.py
class Solution:
    def isStrobogrammatic(self, num: str) -> bool:
        pairs = {'0':'0', '1':'1', '6':'9', '8':'8', '9':'6'}
        left = 0
        right = len(num) - 1
        while left <= right:
            if num[left] not in  pairs or num[right] not in pairs:
                return False
            if pairs[num[left]] != num[right]:
                return False
            left += 1
            right -= 1
        return True

File: 2024/246. Strobogrammatic Number.cpp
class Solution {
public:
    unordered_map<char, char> pairs = {
        {'0', '0'}, {'1', '1'}, {'6', '9'}, {'8', '8'}, {'9', '6'}};
    bool isStrobogrammatic(string num) {
        int left = 0, right = num.length() - 1;
        while (left <= right) {
            if (pairs.find(num[left]) == pairs.end() ||
                pairs[num[left]] != num[right]) {
                return false;
            }
            left += 1;
            right -= 1;
        }
        return true;
    }
};